rzup/src/lib.rs

```diff
@@ -11,64 +11,378 @@
+    pub fn with_root<P: Into<PathBuf>>(root: P) -> Result<Self> {
+        let environment = Environment::with_root(root)?;
+        let mut registry = Registry::new(&environment)?;
+        Self::initialize_settings(&environment, &mut registry)?;
+        registry.scan_environment(&environment)?;
+
+        Ok(Self {
+            environment,
+            registry,
+        })
+    }
+
+    fn initialize_settings(environment: &Environment, registry: &mut Registry) -> Result<()> {
+        if !environment.settings_path().exists() {
+            environment.emit(RzupEvent::Debug {
+                message: format!(
+                    "Creating new settings file at {}",
+                    environment.settings_path().display()
+                ),
+            });
         }
+        registry.settings().save(environment)?;
+        Ok(())
+    }
 
+    /// Sets an event handler for receiving notifications about operations.
+    ///
+    /// # Arguments
+    /// * `handler` - Function that will be called for each event
+    pub fn set_event_handler<F>(&mut self, handler: F)
+    where
+        F: Fn(RzupEvent) + Send + Sync + 'static,
+    {
+        self.environment.set_event_handler(handler);
+    }
+
+    /// Installs all default components.
+    ///
+    /// # Arguments
+    /// * `force` - If true, reinstalls even if already installed
+    pub fn install_all(&mut self, force: bool) -> Result<()> {
+        self.registry
+            .install_all_components(&self.environment, force)?;
+        self.registry.scan_environment(&self.environment)?;
         Ok(())
     }
 
-    pub fn uninstall() -> Result<()> {
-        let cargo_bin_path = dirs::home_dir()
-            .ok_or_else(|| anyhow!("Could not determine home directory"))?
-            .join(".cargo/bin");
+    /// Installs a specific component version.
+    ///
+    /// # Arguments
+    /// * `component` - Component identifier (e.g. "rust", "cargo-risczero")
+    /// * `version` - Specific version to install, or None for latest
+    /// * `force` - If true, reinstalls even if already installed
+    pub fn install_component(
+        &mut self,
+        component: &str,
+        version: Option<Version>,
+        force: bool,
+    ) -> Result<()> {
+        self.registry
+            .install_component(&self.environment, component, version, force)?;
+        self.registry.scan_environment(&self.environment)?;
+        Ok(())
+    }
+
+    /// Uninstalls a specific component version.
+    ///
+    /// # Arguments
+    /// * `component` - Component identifier
+    /// * `version` - Version to uninstall
+    pub fn uninstall_component(&mut self, component: &str, version: Version) -> Result<()> {
+        self.registry
+            .uninstall_component(&self.environment, component, version)?;
+        self.registry.scan_environment(&self.environment)?;
+        Ok(())
+    }
+
+    /// Lists all installed versions of a component.
+    ///
```

Comment:
Maybe it would be good to mention in the doc-comment that the return listed is sorted from newest to oldest